src/js/_enqueues/lib/emoji-loader.js

```diff
@@ -22,438 +31,415 @@
-			resolve( supportTests );
-			return;
+			if ( 'flag' !== test ) {
+				settings.supports.everythingExceptFlag =
+					settings.supports.everythingExceptFlag &&
+					settings.supports[ test ];
+			}
 		}
 
-		if ( supportsWorkerOffloading() ) {
-			try {
-				// Note that the functions are being passed as arguments due to minification.
-				var workerScript =
-					'postMessage(' +
-					testEmojiSupports.toString() +
-					'(' +
-					[
-						JSON.stringify( tests ),
-						browserSupportsEmoji.toString(),
-						emojiSetsRenderIdentically.toString(),
-						emojiRendersEmptyCenterPoint.toString()
-					].join( ',' ) +
-					'));';
-				var blob = new Blob( [ workerScript ], {
-					type: 'text/javascript'
-				} );
-				var worker = new Worker( URL.createObjectURL( blob ), { name: 'wpTestEmojiSupports' } );
-				worker.onmessage = function ( event ) {
-					supportTests = event.data;
-					setSessionSupportTests( supportTests );
-					worker.terminate();
-					resolve( supportTests );
-				};
-				return;
-			} catch ( e ) {}
-		}
+		settings.supports.everythingExceptFlag =
+			settings.supports.everythingExceptFlag &&
+			! settings.supports.flag;
 
-		supportTests = testEmojiSupports( tests, browserSupportsEmoji, emojiSetsRenderIdentically, emojiRendersEmptyCenterPoint );
-		setSessionSupportTests( supportTests );
-		resolve( supportTests );
+		// Sets DOMReady to false and assigns a ready function to settings.
+		settings.DOMReady = false;
+		settings.readyCallback = () => {
+			settings.DOMReady = true;
+		};
 	} )
-		// Once the browser emoji support has been obtained from the session, finalize the settings.
-		.then( function ( supportTests ) {
-			/*
-			 * Tests the browser support for flag emojis and other emojis, and adjusts the
-			 * support settings accordingly.
-			 */
-			for ( var test in supportTests ) {
-				settings.supports[ test ] = supportTests[ test ];
-
-				settings.supports.everything =
-					settings.supports.everything && settings.supports[ test ];
-
-				if ( 'flag' !== test ) {
-					settings.supports.everythingExceptFlag =
-						settings.supports.everythingExceptFlag &&
-						settings.supports[ test ];
-				}
-			}
-
-			settings.supports.everythingExceptFlag =
-				settings.supports.everythingExceptFlag &&
-				! settings.supports.flag;
-
-			// Sets DOMReady to false and assigns a ready function to settings.
-			settings.DOMReady = false;
-			settings.readyCallback = function () {
-				settings.DOMReady = true;
-			};
-		} )
-		.then( function () {
-			return domReadyPromise;
-		} )
-		.then( function () {
-			// When the browser can not render everything we need to load a polyfill.
-			if ( ! settings.supports.everything ) {
-				settings.readyCallback();
-
-				var src = settings.source || {};
-
-				if ( src.concatemoji ) {
-					addScript( src.concatemoji );
-				} else if ( src.wpemoji && src.twemoji ) {
-					addScript( src.twemoji );
-					addScript( src.wpemoji );
-				}
+	.then( () => {
```

Comment:
This secondary `then()` can be removed since there is no intervening `domReadyPromise` to wait to resolve.

Reply:
Addressed in https://github.com/WordPress/wordpress-develop/pull/10145/commits/5afbdcb5ba0831118c5f6a722b13a68c91947950 as part of follow-up PR: https://github.com/WordPress/wordpress-develop/pull/10145